examples/CMakeLists.txt

```diff
@@ -0,0 +1,23 @@
+cmake_minimum_required(VERSION 3.10)
```

Comment:
I think the intention here is to allow users to run all of the examples at once by configuring this directory?

I think this is unnecessary -- Adding a `CCCL_ENABLE_EXAMPLES` option to the top level CCCL CMakeLists.txt that conditionally does `add_subdirectory(examples)` will let us check that the examples are working in CI, and if users want to try out a single example, they can just configure those directly (without even configuring CCCL), which would be more illustrative as an example than configuring this multi-example harness.